File: src/main/java/ru/romanow/scc/warehouse/repository/ItemsRepository.java
package ru.romanow.scc.warehouse.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import ru.romanow.scc.warehouse.domain.Items;

import java.util.List;
import java.util.UUID;

public interface ItemsRepository
        extends JpaRepository<Items, Integer> {

    @Query("select it from Items it where it.uid in :itemsUids")
    List<Items> findByUids(@Param("itemsUids") List<UUID> itemsUid);
}

File: src/main/java/ru/romanow/warehouse/repository/ItemsRepository.kt
package ru.romanow.warehouse.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import ru.romanow.warehouse.domain.Items
import java.util.*

interface ItemsRepository : JpaRepository<Items, Int>, JpaSpecificationExecutor<Items> {

    @Query("select it from Items it where it.uid in :itemsUids")
    fun findByUids(@Param("itemsUids") itemsUid: List<UUID>): List<Items>
}